File: SAV_Frontend/src/Models/Reclamation.ts
import { Article } from "./Article";
import { Client } from "./Client";
import { Intervention } from "./Intervention";
import { StatutReclamation } from "./StatutReclamation";

export interface Reclamation {
    id: number;
    dateReclamation: Date;
    description: string;
    clientId: number;
  
    client?: Client; // Remplacez 'Client' par l'interface appropriée si elle existe
    articleId: number;
  
    article?: Article; // Remplacez 'Article' par l'interface appropriée si elle existe
  
    statutReclamationId: number;
    statutReclamation?: StatutReclamation; // Remplacez 'StatutReclamation' par l'interface appropriée si elle existe
    interventions?: Intervention[]; // Remplacez 'Intervention' par l'interface appropriée si elle existe
  }
  

File: SAV_Backend/SAV_Backend/Models/Reclamation.cs
using System.Text.Json.Serialization;

namespace SAV_Backend.Models
{
    public class Reclamation
    {
        public int Id { get; set; }
        public DateTime DateReclamation { get; set; }
        public string Description { get; set; }
        public int ClientId { get; set; }
        [JsonIgnore]
        public Client?Client { get; set; }
        public int? ArticleId { get; set; }

        [JsonIgnore]
        public Article? Article { get; set; }
                                        
        public int StatutReclamationId { get; set; }
        public StatutReclamation? StatutReclamation { get; set; } 
        public ICollection<Intervention>? Interventions { get; set; }
    }



}
